cmd/end2endtest/helpers.go

```diff
@@ -508,11 +507,13 @@ func ballotVotes(b ballotData, nvotes int) ([][]int, [][]*types.BigInt) {
 
 // sendVotes sends a batch of votes concurrently
 // (number of goroutines defined in t.config.parallelCount)
-func (t *e2eElection) sendVotes(votes []*apiclient.VoteData) (errs map[int]error) {
-	errs = make(map[int]error)
-	var timeouts atomic.Uint32
+func (t *e2eElection) sendVotes(votes []*apiclient.VoteData) []error {
+	// used to avoid infinite for loop
+	var timeoutsRetry, waitRetry, warnRetry atomic.Uint32
 	var wg sync.WaitGroup
 	var queues []map[int]*apiclient.VoteData
+
+	errs := []error{}
```

Comment:
slice errs is not concurrent safe (and you're reading/writing it in concurrent goroutines)

Reply:
i must say that my code used a map, that is also not concurrent safe. so the current code is buggy as well, sorry about that.

Reply:
updated